main/src/com/google/refine/importing/ImportingUtilities.java

```diff
@@ -846,6 +850,22 @@ public int compare(String o1, String o2) {
         
         return formats.size() > 0 ? formats.get(0) : null;
     }
+
+    static protected String guessFormatFromMimeType(String mimeType) {
+        if (mimeType == null) {
+            return null;
+        }
+
+        if (mimeType.contains("/xml")) {
+            return "text/xml";
+        }
+
+        if (mimeType.contains("/json")) {
+            return "text/json";
+        }
+
```

Comment:
It seems to me that this function is duplicating functionality that should already be implemented elsewhere, since we are already storing a mapping from MIME types to formats in `main/webapp/modules/core/MOD-INF/controller.js`.

Could the issue be solved simply by adding the relevant MIME types in `controller.js`? If patterns are necessary, such as `.*/xml.*`, we could consider adding support for patterns in this MIME type registry.

Reply:
@wetneb this is really a good idea. update it now